bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandler.java

```diff
@@ -174,4 +175,24 @@
      */
     void handleRemoval();
 
+    /**
+     * This method should return a unique identifier of the device handled by this {@link ThingHandler}.
```

Comment:
Not sure, if I understood you right, introducing getRepresentationProperty() would not help as the method to retrieve the unique device id is different in each handler and not necessarily only getting some device property, because in the manual case the properties might also been set manually. I agree we should not use device here, so what about `getUniqueIdentifier()`or `getRepresentationValue()`?